### DISCUSSION TITLE
Possible to override tokenizer at spacy.load function via config?

### DISCUSSION THREAD

**Discussion by kinghuang** [explosion/spaCy#12618] [Help: Best practices]
The [spacy.load](https://spacy.io/api/top-level#spacy.load) function takes a `config` argument that allows the model config to be tweaked at load time. Is there a way to override the `nlp.tokenizer` config? I'm trying to do the following, but getting an error that the section value cannot be overwritten.

```python
nlp = spacy.load("en_tsds_doc_maker", config={"nlp.tokenizer": {"@tokenizers":"oi.Tokenizer.v1"}})
```

```
confection.ConfigValidationError: 

Error parsing config overrides
nlp -> tokenizer -> @tokenizers	not a section value that can be overwritten
```

What I want to do is assign `{"@tokenizers":"oi.Tokenizer.v1"}` to `nlp.tokenizer`. But, it appears to be trying to assign `oi.Tokenizer.v1` to `nlp.tokenizer.@tokenizers`. Is there some way to make this work?

**Comment by bdura** [comment]
Hello @kinghuang, that's a good question - we'll need to look into what exactly is happening there.

However, in general overriding the tokenizer configuration is not a great idea because it's unlikely you'll be able to load the model in `from_disk`.

**Reply by kinghuang** [reply]
Thanks. In this case, the alternate tokenizer is packaged with the model. I can build two separate model packages, but I'd rather build one and just swap out the tokenizer on the fly via config.

**Comment by kinghuang** [comment]
Fixed in #12623.
